### PR TITLE
feat(protocol-impl): add tenant ids property for job poll and push requests

### LINKED ISSUE #13318: `JobBatchRecord` and `JobActivationProperties` provide tenant information
### Description

To support multi-tenancy when polling or pushing jobs, a job should be activated when the `jobType` **AND** `tenantId` match.
We also want to be able to poll or push jobs for multiple tenants at once.

As a result, a list of `tenantIds` needs to be included in the `JobBatchRecord`s and `JobActivationProperties` classes, to ensure that filtering jobs by `jobType` **AND** `tenantId` is possible.

```[tasklist]
### AC
- [x] The `JobBatchRecord` class provides a `tenantIds` property (List\<String\>).
- [x] The `JobActivationProperties` class provides a `tenantIds` property (List\<String\>).
- [x] ~The `JobBatchRecord` class implements the `RecordValueWithTenantPermissions` interface~ (see #13989 and #14283)
- [x] ~The `JobActivationProperties` class implements the `RecordValueWithTenantPermissions` interface~ (see #13989 and #14283)
- [x] ~The Gateway adds a list of tenant ids assigned to a user to the`JobBatchRecord` or `JobActivationProperties` instances for each `ActivateJobs`/`StreamJobs` RPC call.~ (see #14283)
```

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Implement the `tenantIds` property for the `JobBatchRecord` and the `JobActivationProperties` classes. The new `tenantIds` property will declare the IDs of the tenants that users want to activate and process jobs for.

## Related issues

closes #13318 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/stream/job/JobActivationProperties.java
Comment (korthout) [inline protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/stream/job/JobActivationProperties.java:51]: 🔧 Instead of focusing on how jobs streamed from the engine could belong to different tenants, I suggest we flip this around to activating as an action, instead of something that the engine does for you. This fits the context of this class better, which is about the properties of the jobs to activate for a stream that is being opened.

> A stream can be opened to activate the jobs of multiple tenants. Each activated job will be owned by the tenant that owns the corresponding process instance.

**Review by korthout — APPROVED** [review]
Thanks @koevskinikola 🚀 

LGTM 👍 

PS: Sorry for the late review 🙇

File: protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/stream/job/JobActivationProperties.java
Comment (koevskinikola) [reply protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/stream/job/JobActivationProperties.java]: @korthout I rephrased the javadoc. To make faster progress, I'll merge this PR. In case you have more feedback, I can implement the changes in later multi-tenancy-related PRs.

Comment (korthout) [reply protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/stream/job/JobActivationProperties.java]: 💭 @koevskinikola Please feel free to merge after I've approved a pull request and when you make some changes due to suggestions unless you feel those changes need a second pair of eyes. In cases where I feel I want to have another look, I'll typically reject the changes to force another review.

**Comment by koevskinikola** [comment]
bors merge

**Pull request merged**
